lib/core/src/msParam.cpp

```diff
@@ -1451,60 +1451,61 @@ int
 addKeyValToMspStr( msParam_t * keyStr, msParam_t * valStr,
                    msParam_t * msKeyValStr ) {
 
-    if ( ( keyStr == NULL && valStr == NULL ) || msKeyValStr == NULL ) {
+    if ((keyStr == nullptr && valStr == nullptr) || msKeyValStr == nullptr) {
```

Comment:
What happens if `valStr` is valid, but `keyStr` is null?
Do we protect against that case later in the function?

Reply:
I believe both `parseMspForStr()` and the lambda `get_length()` handle the `nullptr` case. Whether we want a value but no key is a different discussion: https://github.com/irods/irods/pull/7049#discussion_r1171576383.